server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -2113,12 +2113,14 @@ export class AgenticChatController implements ChatHandlers {
         metric.metric.cwsprChatConversationId = conversationId
         await this.#telemetryController.emitAddMessageMetric(tabId, metric.metric, 'Failed')
 
-        if (isFreeTierLimitError(err)) {
-            this.setPaidTierMode(tabId, 'freetier-limit')
+        if (isUsageLimitError(err)) {
+            if (this.#paidTierMode !== 'paidtier') {
+                this.setPaidTierMode(tabId, 'freetier-limit')
+            }
             return new ResponseError<ChatResult>(LSPErrorCodes.RequestFailed, err.message, {
                 type: 'answer',
-                body: `AmazonQFreeTierLimitError: Free tier limit reached. ${requestID ? `\n\nRequest ID: ${requestID}` : ''}`,
-                messageId: 'freetier-limit',
+                body: `AmazonQUsageLimitError: Monthly limit reached. ${requestID ? `\n\nRequest ID: ${requestID}` : ''}`,
```

Comment:
Reword messages to reflect that the monthly limit is not strictly for "free tier".